scalding-commons-macros/src/main/scala/com/twitter/scalding/commons/macros/impl/ordser/ScroogeOuterOrderedBuf.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.commons.macros.impl.ordser
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+import com.twitter.scalding._
+import java.nio.ByteBuffer
+import com.twitter.scalding.serialization.OrderedSerialization
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+import com.twitter.scrooge.{ ThriftUnion, ThriftStruct }
+import com.twitter.scalding.macros.impl.ordser._
+
+object ScroogeOuterOrderedBuf {
```

Comment:
can you add a comment as to what is handled here? It looks like ThriftStruct that is not a ThriftUnion.